### PR TITLE
polynomial_ring cached=false for combinatorics

### PR DESCRIPTION
**Pull request by ederc**
Some tests have now problems with different parent rings, see discussion at #2455.

Maybe @bschroter can help fixing the tests?

### CODE REVIEW

File: src/Combinatorics/Matroids/properties.jl [hunk 1]
```diff
@@ -918,7 +918,7 @@ q^3 - 7*q^2 + 14*q - 8
 ```
 """
 function characteristic_polynomial(M::Matroid)
```
Comment (fingolfin) [inline src/Combinatorics/Matroids/properties.jl]: These functions touched here should all at some point be changed allow passing in the polynomial ring, as discussed [here in our dev docs](https://docs.oscar-system.org/dev/DeveloperDocumentation/styleguide/#Optional-arguments-for-parents-of-return-values)

(To be clear, that's not a request to change this PR)

File: src/Combinatorics/Matroids/properties.jl [hunk 2]
```diff
@@ -936,7 +936,7 @@ q^2 - 6*q + 8
 ```
 """
 function reduced_characteristic_polynomial(M::Matroid)
-    R, q = polynomial_ring(ZZ, 'q')
+    R, q = polynomial_ring(ZZ, 'q'; cached=false)
     p = characteristic_polynomial(M)
```
Comment (fingolfin) [inline src/Combinatorics/Matroids/properties.jl]:
```suggestion
    p = characteristic_polynomial(M)
    R = parent(p)
```

File: src/Combinatorics/Matroids/properties.jl [hunk 3]
```diff
@@ -936,7 +936,7 @@ q^2 - 6*q + 8
...
     c = Vector{Int}(undef,degree(p))
```
Comment (fingolfin) [inline src/Combinatorics/Matroids/properties.jl]: Huh... that seems risky, couldn't one have examples where an `Int` would over- or underflow?

File: src/Combinatorics/Matroids/properties.jl [hunk 4]
```diff
@@ -884,7 +884,8 @@ function tutte_connectivity(M::Matroid)
 end
 
 @doc raw"""
-    tutte_polynomial(M::Matroid)
+    tutte_polynomial(M::Matroid; parent::ZZMPolyRing)
```
Comment (fingolfin) [inline src/Combinatorics/Matroids/properties.jl:888]: Can we indicate somehow that `parent` is optional? E.g.
```suggestion
    tutte_polynomial(M::Matroid[; parent::ZZMPolyRing])
```
but I am not sure this is universally understood. Or perhaps
```suggestion
    tutte_polynomial(M::Matroid; parent::ZZMPolyRing = ...)
```
and/or just say in the text that a parent can be *optionally* passed via kwarg or as first argument or so

**Review by lgoettgens — APPROVED** [review]

**Pull request merged**